lib/cloudex/cloudinary_api.ex

```diff
@@ -92,53 +92,34 @@ defmodule Cloudex.CloudinaryApi do
   @spec upload_file(String.t(), map) ::
           {:ok, %Cloudex.UploadedImage{}} | {:ok, %Cloudex.UploadedVideo{}} | {:error, any}
   defp upload_file(file_path, opts) do
-    poison_opts = get_httpoison(opts)
-
     options =
       opts
+      |> Map.delete(:request_options)
```

Comment:
I still feel like we need to delete the request_options key so that it can make the body.

Reply:
opts will not be changed so the key will still exist when passed through post(body, file_path, opts)

Reply:
Shouldn't we follow the same pattern with `Map.pop/3` here?

Reply:
@scrogson No I don't think we should use Map.pop here. Because it wouldn't go through the pipeline. It gives back the {default/value, new map} and deleting gives back the map only.